### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by yumuzi234**

### CODE REVIEW

File: errcode/error.go [hunk 1]
```diff
@@ -55,6 +56,11 @@ func IsUnauthorized(err error) bool {
 	return Of(err) == Unauthorized
 }
 
+// IsTimeOut checks if it is an time-out error.
+func IsTimeOut(err error) bool {
```
Comment (h8liu) [inline errcode/error.go]: where do you need this?

File: idutil/short_test.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package idutil
```
Comment (h8liu) [inline idutil/short_test.go]: this file is merged; please pull and sync.

File: jsonfile/jsonfile_test.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
+package jsonfile
```
Comment (h8liu) [inline jsonfile/jsonfile_test.go]: we should use `testdata` folder here.

File: moment/moment.go [hunk 4]
```diff
@@ -14,7 +14,8 @@ func ForNow(t time.Time) string {
 // String returns a human friendly string for describing a time point t
 // relative to now.
 func String(t, now time.Time) string {
```
Comment (h8liu) [inline moment/moment.go]: per discussion, need to sync to timezone in `now`

File: strutil/strutil_test.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
+package strutil
```
Comment (h8liu) [inline strutil/strutil_test.go]: this file is merged; please pull and sync.

File: signer/rsa_time_signer_test.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+import (
+	"testing"
+	"time"
+
+	"crypto/rand"
+	"crypto/rsa"
+)
+
+func TestRsaTimeSigner(t *testing.T) {
```
Comment (h8liu) [inline signer/rsa_time_signer_test.go]: `TestRSATimeSigner`. RSA should be capital.

File: errcode/error.go (hunk 1 above)
Comment (yumuzi234) [reply errcode/error.go]: I added this when dealing with RSASigner, also you mentioned for Context, there should be a deadline. So I added it.

File: jsonfile/jsonfile_test.go (hunk 3 above)
Comment (yumuzi234) [reply jsonfile/jsonfile_test.go]: You mean create files under testdata folder, and try to read/write from there?
Basically I copied the test from
https://golang.org/src/io/ioutil/ioutil_test.go
Line 45 is quite funny...

File: moment/moment.go [hunk 7]
```diff
@@ -6,15 +6,16 @@ import (
...
 // ForNow returns a human friendly string for describing a time point
-// t relative to now. It reads the system's current time.
+// t relative to current time. It reads the system's current time.
 func ForNow(t time.Time) string {
 	return String(t, time.Now())
 }
 
 // String returns a human friendly string for describing a time point t
-// relative to now.
+// relative to `now`. The time zone will be sync to timezone in `now` first.
```
Comment (h8liu) [inline moment/moment.go]: don't quote variables, go dos do not do that. if you think it is not clear, rewrite it.

File: jsonfile/jsonfile_test.go [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+import (
+	"io/ioutil"
+	"os"
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (yumuzi234) [reply jsonfile/jsonfile_test.go]: I have no idea. My local branch is OK, and git pull/push will not change anything, says everything is up-to-date

File: errcode/error.go (hunk 1 above)
Comment (h8liu) [reply errcode/error.go]: rsa signer should generate something like token-expired error. it is not a time-out.

time-out means that an operation takes more time that expected.

File: signer/time.go [hunk 9]
```diff
@@ -13,6 +13,7 @@ func now(f func() time.Time) time.Time {
 	return f()
 }
 
+// Is this what we want? The duration is actually 2w here
```
Comment (h8liu) [inline signer/time.go]: yes... windows is for tolerating time differences among servers and clients.

Comment (h8liu) [reply signer/time.go]: don't ever put these kind of questions into the code of a pull request. ask them in the discussion of a pull request or anywhere else. 

try to make every pull request to be "ready-to-merge".

we want to build a temple of elegant go language code, and serve as priests. so please treat code as something sacred.

Comment (yumuzi234) [reply signer/time.go]: my bad

**Pull request merged**